builder/evaluator.go

```diff
@@ -191,6 +197,18 @@ func (b *builder) Run(context io.Reader) (string, error) {
 		}
 	}
 
+	// check if there are any leftover build-args that were passed but not
+	// consumed during build. Return an error, if there are any.
+	leftoverArgs := []string{}
```

Comment:
We should pre-parse the Dockerfile rather than waiting until the end... but I suppose this is an optimization.